cli/azd/internal/agent/agent_factory.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package agent
+
+import (
+	"github.com/azure/azure-dev/cli/azd/internal/agent/consent"
+	"github.com/azure/azure-dev/cli/azd/internal/agent/logging"
+	localtools "github.com/azure/azure-dev/cli/azd/internal/agent/tools"
+	"github.com/azure/azure-dev/cli/azd/internal/agent/tools/common"
+	mcptools "github.com/azure/azure-dev/cli/azd/internal/agent/tools/mcp"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
+	"github.com/azure/azure-dev/cli/azd/pkg/llm"
+)
+
+type AgentFactory struct {
+	consentManager consent.ConsentManager
+	llmManager     *llm.Manager
+	console        input.Console
+}
+
+func NewAgentFactory(
+	consentManager consent.ConsentManager,
+	console input.Console,
+	llmManager *llm.Manager,
+) *AgentFactory {
+	return &AgentFactory{
+		consentManager: consentManager,
+		llmManager:     llmManager,
+		console:        console,
+	}
+}
+
+func (f *AgentFactory) Create(opts ...AgentOption) (Agent, func() error, error) {
```

Comment:
Should we call the options `CreateOptions` ?   We usually follow the `<FunctionName>Options` convention for naming options.